### PR TITLE
Packet handlers

### PR DESCRIPTION
**Pull request by ahkneipp**
Packet unpackers and unit tests

### CODE REVIEW

File: doc/.gitignore [hunk 1]
```diff
@@ -10,4 +10,4 @@
 *.glo
 *.gls
 *.ist
-
+fixes.txt
```
Comment (thatging3rkid) [inline doc/.gitignore:13]: still necessary?

File: network/netutils.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        case kv_Boolean:
+            value.Boolean = *((int8_t*)(valStart));
+            break;
+    }
+    *end = valStart+valueLength;
+    return KVPairTLV_create(key, type, valueLength+4, value);
+}
+
+///Check if two KVPairTLV's are equal
+int KVPairTLV_equals(KVPairTLV_t* kv1, KVPairTLV_t* kv2)
```
Comment (thatging3rkid) [inline network/netutils.c]: would prefer to see a bool over int unless we're planning to return a non-0-or-1 value i.e. 
```c
for (size_t i =0 ; i < min(kv1->length, kv2->length); i += 1) {
    int diff = kv1->value.CString[i] - kv2->value.CString[i];
    if (diff != 0) {
        return diff;
    }
}
```

File: network/packethandlers.c [hunk 3]
```diff
@@ -2,4 +2,598 @@
...
+    {
+        //packet allocation succeeded, so free that memory
+        free(packet);
+        //Free the raw packet
+        llnet_packet_free(rawPacket);
+        return NULL;
+    }
+    packet->type = rawPacket->type;
+    packet->timestamp = rawPacket->timestamp;
+    packet->length = rawPacket->length;
```
Comment (thatging3rkid) [inline network/packethandlers.c]: isn't this unnecessary, done by `createBasePacket(...)`?

File: network/packethandlers.c [hunk 4]
```diff
@@ -2,4 +2,598 @@
...
+    }
+
+    packet->data = (PTLVData_Base_t*)unpacked;
+
+    unpacked->left_stick_x = rawPacket->data[0];
+    unpacked->left_stick_y = rawPacket->data[1];
+    unpacked->right_stick_x = rawPacket->data[2];
+    unpacked->right_stick_y = rawPacket->data[3];
+    unpacked->button_a = (rawPacket->data[4] & 0x80) != 0;
+    unpacked->button_b = (rawPacket->data[4] & 0x40) != 0;
```
Comment (thatging3rkid) [inline network/packethandlers.c:380]: what about buttons 3 and 4?

File: network/packethandlers.c [hunk 5]
```diff
@@ -2,4 +2,598 @@
...
+        return NULL;
+    }
+
+    packet->data = (PTLVData_Base_t*)unpacked;
+
+    unpacked->code_status = (rawPacket->data[0] & 0xF0) >> 4;
+    unpacked->commit_hash = (rawPacket->data[0] & 0x0F) << 24 | rawPacket->data[1] << 16| rawPacket->data[2] << 8 | rawPacket->data[3];
+    unpacked->robot_uuid = ((uint32_t*)(rawPacket->data))[1];
+    unpacked->state = (RobotState_t)(rawPacket->data[8]);
+    unpacked->config_entries = ((uint16_t*)(rawPacket->data))[5];
```
Comment (thatging3rkid) [inline network/packethandlers.c]: I'm not a huge fan of magic number indexes, I'd rather see
```c
// note: not tested code
uint8_t* data_ptr= rawPacket->data;
unpacked->code_status = (data_ptr[0] & 0xf0) >> 4;
unpacked->commit_hash = ((data_ptr[0] & 0x0f) << 24) | (data_ptr[1] << 16) | (data_ptr[2] << 8) | (data_ptr[3] << 0);
data_ptr += 4;
unpacked->robot_uuid = ((uint32_t*) data_ptr)[0];
data_ptr += 4;
...
```

**Review by thatging3rkid — APPROVED** [review]
looks good, I just have some pedantic comments

File: doc/.gitignore (hunk 1 above)
Comment (ahkneipp) [reply doc/.gitignore:13]: Yes, I'm storing changes to the protocol definition there until I can do a batch update or clean them up and put them in the repo

File: network/netutils.c (hunk 2 above)
Comment (ahkneipp) [reply network/netutils.c]: Changed the type but didn't update the return statements, shouldn't be a big deal

Comment (ahkneipp) [reply network/netutils.c]: Nevermind, shortened function with bool shit

File: network/packethandlers.c (hunk 3 above)
Comment (ahkneipp) [reply network/packethandlers.c]: Well would you look at that

File: network/packethandlers.c (hunk 4 above)
Comment (ahkneipp) [reply network/packethandlers.c:380]: Only A and B are implemented in the standard, but it's right next to a reserved block.  I'll expand it in the next update for the protocol

File: network/packethandlers.c (hunk 5 above)
Comment (ahkneipp) [reply network/packethandlers.c]: Somewhat cleaner now

**Review by thatging3rkid — APPROVED** [review]
looks good!

**Pull request merged**